### PR TITLE
Fix argument names in torch doc

### LINKED ISSUE #8698: [docs] Argument names in docs don't match those actually taken by function
## Issue description
A bunch of functions have this behaviour. Please check the script below. The functions in the script might not be exhaustive.

EDIT: Around 40 functions have this issue.
```
Exception message: addmm() missing 3 required positional argument: "input", "mat1", "mat2"
Given argument names: ['mat', 'mat1', 'mat2']
Exception message: addmv() missing 3 required positional argument: "input", "mat", "vec"
Given argument names: ['tensor', 'mat', 'vec']
Exception message: addr() missing 3 required positional argument: "input", "vec1", "vec2"
Given argument names: ['mat', 'vec1', 'vec2']
Exception message: baddbmm() missing 3 required positional argument: "input", "batch1", "batch2"
Given argument names: ['mat', 'batch1', 'batch2']
Exception message: bmm() missing 2 required positional argument: "input", "mat2"
Given argument names: ['batch1', 'batch2']
Exception message: dot() missing 2 required positional argument: "input", "tensor"
Given argument names: ['tensor1', 'tensor2']
Exception message: ger() missing 2 required positional argument: "input", "vec2"
Given argument names: ['vec1', 'vec2']
Exception message: matmul() missing 2 required positional argument: "input", "other"
Given argument names: ['tensor1', 'tensor2']
Exception message: mm() missing 2 required positional argument: "input", "mat2"
Given argument names: ['mat1', 'mat2']
Exception message: mv() missing 2 required positional argument: "input", "vec"
Given argument names: ['mat', 'vec']
```

## Code example
```
import torch

def test_function(func, inputs):
    try:
        func(**inputs)
        print("No error with function {}".format(func.__name__))
    except Exception as e:
        print("Exception message: {}\nGiven argument names: {}".format(e, list(inputs.keys())))

if __name__ == '__main__':
    cases = [
             ('addmm', {'mat': torch.randn(2, 4), 'mat1': torch.randn(3, 2, 3), 'mat2': torch.randn(3, 3, 4)}),
             ('addmv', {'tensor': torch.randn(4), 'mat': torch.randn(3, 4, 2), 'vec': torch.randn(3, 2)}),
             ('addr', {'mat': torch.randn(2, 4), 'vec1': torch.randn(3, 2), 'vec2': torch.randn(3, 4)}),
             ('baddbmm', {'mat': torch.randn(3, 2, 4), 'batch1': torch.randn(3, 2, 3), 'batch2': torch.randn(3, 3, 4)}),
             ('bmm', {'batch1': torch.randn(3, 2, 3), 'batch2': torch.randn(3, 3, 4)}),
             ('dot', {'tensor1': torch.randn(10), 'tensor2': torch.randn(10)}),
             ('ger', {'vec1': torch.randn(2), 'vec2': torch.randn(4)}),
             ('matmul', {'tensor1': torch.randn(3, 3), 'tensor2': torch.randn(3, 4)}),
             ('mm', {'mat1': torch.randn(3, 3), 'mat2': torch.randn(3, 4)}),
             ('mv', {'mat': torch.randn(3, 3), 'vec': torch.randn(3)}),
            ]
    for case in cases:
        test_function(getattr(torch, case[0]), case[1])
```
## System Info
- PyTorch or Caffe2: PyTorch
- How you installed PyTorch (conda, pip, source): source (5ca4f5b)

### PR DESCRIPTION
**Pull request by yf225**
I manually went through all functions in `torch.*` and corrected any mismatch between the arguments mentioned in doc and the ones actually taken by the function. This fixes https://github.com/pytorch/pytorch/issues/8698.

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -928,36 +912,36 @@ def merge_dicts(*dicts):
 """)
 
 add_docstr(torch.cholesky_solve, r"""
-cholesky_solve(b, u, upper=False, out=None) -> Tensor
+cholesky_solve(input, input2, upper=False, out=None) -> Tensor
 
 Solves a linear system of equations with a positive semidefinite
-matrix to be inverted given its Cholesky factor matrix :attr:`u`.
+matrix to be inverted given its Cholesky factor matrix :attr:`input2`.
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: Could we create an alias for input2 as u and input as b? I feel that it doesn’t look as good in math expressions.

For example:
```..... its Cholesky factor matrix :math:`u` (given by :attr:`input2`) ....```

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -989,26 +973,26 @@ def merge_dicts(*dicts):
...
+Cholesky factor :attr:`input`: returns matrix ``inv``. The inverse is computed using
 LAPACK routines ``dpotri`` and ``spotri`` (and the corresponding MAGMA routines).
 
-If :attr:`upper` is ``False``, :attr:`u` is lower triangular
+If :attr:`upper` is ``False``, :attr:`input` is lower triangular
 such that the returned tensor is
 
 .. math::
-    inv = (uu^{T})^{-1}
+    inv = (input input^{T})^{-1}
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: I am referring to the same point as in my earlier comment here.

**Review by vishwakftw — COMMENTED** [review]
I am just highlighting a few places where aliases to the argument name could make the math look neater in my opinion. If you like, I could help point out all the other places as well.

**Comment by zou3519** [comment]
> I am just highlighting a few places where aliases to the argument name could make the math look neater in my opinion. If you like, I could help point out all the other places as well.

I agree with this point; using aliases where helpful can make the math look a lot more neat and readable.

**Comment by yf225** [comment]
@vishwakftw @zou3519 I added aliasing in the `Args:` list where that's helpful for making the math formula more clear. Would you like to have a second look at the changes?

File: torch/_torch_docs.py [hunk 3]
```diff
@@ -3786,19 +3770,19 @@ def merge_dicts(*dicts):
...
-:attr:`base` is a scalar ``float`` value, and :attr:`input` is a tensor.
-The returned tensor :attr:`out` is of the same shape as :attr:`input`
+:attr:`self` is a scalar ``float`` value, and :attr:`exponent` is a tensor.
+The returned tensor :attr:`out` is of the same shape as :attr:`exponent`
 
 The operation applied is:
 
 .. math::
-    out_i = base ^ {input_i}
+    out_i = self ^ {exponent_i}
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: Nit: \text{out}_i = \text{self} ^ {\text{exponent}_i}

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -5875,28 +5859,28 @@ def merge_dicts(*dicts):
...
-.. function:: where(condition, x, y) -> Tensor
+.. function:: where(condition, input, other) -> Tensor
 
-Return a tensor of elements selected from either :attr:`x` or :attr:`y`, depending on :attr:`condition`.
+Return a tensor of elements selected from either :attr:`input` or :attr:`other`, depending on :attr:`condition`.
 
 The operation is defined as:
 
 .. math::
     out_i = \begin{cases}
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: Nit: \text{out}_i

File: torch/_torch_docs.py [hunk 5]
```diff
@@ -5875,28 +5859,28 @@ def merge_dicts(*dicts):
...
-        x_i & \text{if } \text{condition}_i \\
-        y_i & \text{otherwise} \\
+        input_i & \text{if } \text{condition}_i \\
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: Nit: \text{input}_i

File: torch/_torch_docs.py [hunk 6]
```diff
@@ -5875,28 +5859,28 @@ def merge_dicts(*dicts):
...
+        other_i & \text{otherwise} \\
```
Comment (vishwakftw) [inline torch/_torch_docs.py]: Nit: \text{other}_i

**Review by vishwakftw — COMMENTED** [review]
Thank you for making the changes. Just a few nits to be corrected.

**Comment by yf225** [comment]
@vishwakftw All fixed, thanks a lot for the suggestions!

**Review by vishwakftw — APPROVED** [review]
All looks good! Thank you for fixing this.

cc: @zou3519

**Review by zou3519 — APPROVED** [review]
Looks fine, but do we have a plan to prevent these from regressing? It would be nice if we could test these doc strings for correctness.

**Comment by vishwakftw** [comment]
@yf225 the test failures seem to be legitimate
```
Jul 18 21:39:52 ======================================================================
Jul 18 21:39:52 FAIL: test_torch (__main__.TestDocCoverage)
Jul 18 21:39:52 ----------------------------------------------------------------------
Jul 18 21:39:52 Traceback (most recent call last):
Jul 18 21:39:52   File "test_docs_coverage.py", line 73, in test_torch
Jul 18 21:39:52     don't want to document?''')
Jul 18 21:39:52 AssertionError: 
Jul 18 21:39:52 The lists of functions documented in torch.rst and in python are different.
Jul 18 21:39:52 Did you forget to add a new thing to torch.rst, or whitelist things you
Jul 18 21:39:52 don't want to document?
```

**Pull request closed without merging**